.github/workflows/tests.yml

```diff
@@ -0,0 +1,684 @@
+
+#     - name: Install tesseract
+#       run: |
+#         sudo apt update
+#         sudo apt-get install tesseract-ocr libtesseract-dev poppler-utils
+#     - name: Install Haystack
+#       run: pip install .
+
+#     - name: Run tests
+#       env:
+#         TOKENIZERS_PARALLELISM: 'false'  # Avoid logspam by tokenizers
+#       run: |
+#         pytest ${{ env.PYTEST_PARAMS }} -m "integration" test/${{ matrix.folder }}
+
+
+#   integration-tests-windows:
+#     needs: 
+#      - unit-tests-windows
+#      - elasticsearch-tests-windows
+#     runs-on: windows-latest
+#     if: contains(github.event.pull_request.labels.*.name, 'topic:windows') || !github.event.pull_request.draft
+
+#     timeout-minutes: 30
+#     strategy:
+#       fail-fast: false  # Avoid cancelling the others if one of these fails
+#       matrix: 
+#         folder:
+#           - "nodes"
+#           - "pipelines"
+#           - "modeling"
+#           - "others"
+
+#     steps:
+#     - uses: actions/checkout@v2
+
+#     - name: Set up Elasticsearch and pdftotext
+#       run: |
+#         choco install xpdf-utils
+#         choco install openjdk11
+#         refreshenv
+#         choco install tesseract --pre
+#         choco install elasticsearch --version=7.9.2
+#         refreshenv
+#         Get-Service elasticsearch-service-x64 | Start-Service
+
+#     - name: Setup Python
+#       uses: ./.github/actions/python_cache/
+#       with:
+#         prefix: windows
+
+#     - name: Install Haystack
+#       run: pip install .
+
+#     - name: Run tests
+#       env:
+#         TOKENIZERS_PARALLELISM: 'false'  # Avoid logspam by tokenizers
+#       # FIXME many tests are disabled here!
+#       run: |
+#         pytest ${{ env.PYTEST_PARAMS }} -m "integration and not tika and not graphdb" ${{ env.SUITES_EXCLUDED_FROM_WINDOWS }} test/${{ matrix.folder }} --document_store_type=memory,faiss,elasticsearch
+ 
+ 
+  # This CI action mirrors autoformat.yml, with the difference that it
+  # runs on Haystack's end. If the contributor hasn't run autoformat.yml, 
+  # then this check will fail.
+  code-and-docs-check:
+    runs-on: ubuntu-latest
+    #needs: integration-tests-linux
+    steps:
+
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+          ref: ${{ github.head_ref }}
+          
+      - name: Setup Python
+        uses: ./.github/actions/python_cache/
+
+      - name: Install Dependencies
+        run: |
+          pip install --upgrade pip
+          pip install .[all]
+          pip install rest_api/
+          pip install ui/
+
+      # Get any additional commit that might have been pushed in the meantime
+      - name: Pull changes (if any)
+        run: git pull origin ${{ github.head_ref }}
```

Comment:
I'm not sure I like this, it breaks the commit<-->workflow relation in Github UI, what's the rationale for this step?

Reply:
And it is also failing, so it will change :smile: 

The idea is the following:
1. The code bot and the tests workflow will start independently on any new PR
2. The code bot will commit on the branch
3. The test workflow will reach the end in the meantime and verify if the bot had to apply changes and, in that case, if it applied them. It does this by: 
   - pulling any new commit that might have been done in the meantime,
   - checking if running the bash script against makes any further changes. if there are further changes, it fails, because it means the code bot did not do its job.
   
I agree this is quite convoluted, but that's the best I could come up with to apply code formatting and code generation on the CI. Radical opinions welcome! We should also evaluate what makes sense to do in this PR though.

Reply:
My bad, I didn't notice the workflow isn't triggered after the docs push, scratch the idea!

Reply:
I will still give an attempt at running the docs check after either the docs update workflow or the tests workflow, whichever finishes first. I realized dependency among workflows is possible! https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run